lib/matrix/matrix.nit

```diff
@@ -293,3 +307,30 @@ class MatrixCoordinate
 
 	redef fun to_s do return "({x},{y})"
 end
+
+# Dedicated native structure to avoid boxing cost
+private extern class NativeMatrix `{ double* `}
+
+	new(buffer: CString) `{ return (double*)buffer; `}
```

Comment:
OK. I think there is two issues here. The first one is the NativeMatrix API. the second one is what is the best way to manage raw allocated memory.

Both issues can be solved independently. For the first, why not moving the pseudo factory in `Matrix::items` into a real factory in NativeMatrix?

~~~nit
private extern class NativeMatrix `{ double* `}
	new(size: Int) do
		var sizeof_double = 8
		var buf = new CString(sizeof_double*size)
		return new NativeMatrix.from_buffer(buf)
	end
	new from_buffer(buffer: CString) `{ return (double*)buffer; `}
	...
~~~

For the second issue, CString seems weird at first, but it does the job.

By the way, `NativeMatrix` is a counter-intuitive name, what you have is a `NativeFloatArray` in fact.